torch/autograd/_functions/tensor.py

```diff
@@ -73,522 +68,498 @@ def _do_forward(self, *args, **kwargs):
+        ctx.save_for_backward(index)
+        ctx.dim = dim
+        return input.gather(dim, index)
 
-    def backward(self, grad_output):
-        index, = self.saved_tensors
-        grad_input = grad_output.new(self.input_size).zero_()
-        return grad_input.scatter_(self.dim, index, grad_output), None
+    @staticmethod
+    @once_differentiable
+    def backward(ctx, grad_output):
+        index, = ctx.saved_tensors
+        grad_input = grad_output.new(ctx.input_size).zero_()
+        return grad_input.scatter_(ctx.dim, index, grad_output), None, None
 
 
 class Scatter(InplaceFunction):
 
-    def __init__(self, dim, inplace=False):
-        super(Scatter, self).__init__(inplace)
-        self.dim = dim
-
-    def forward(self, input, index, source):
-        assert not self.needs_input_grad[1], "Scatter can't differentiate " \
-            "the index"
-        if self.inplace:
-            self.mark_dirty(input)
+    @staticmethod
+    def forward(ctx, input, dim, index, source, inplace=False):
+        assert not ctx.needs_input_grad[2], "Scatter can't differentiate the index"
+        ctx.dim = dim
+        if inplace:
+            ctx.mark_dirty(input)
         else:
             input = input.clone()
-        self.save_for_backward(index)
-        return input.scatter_(self.dim, index, source)
+        ctx.save_for_backward(index)
+        return input.scatter_(ctx.dim, index, source)
 
-    def backward(self, grad_output):
-        index, = self.saved_tensors
+    @staticmethod
+    @once_differentiable
+    def backward(ctx, grad_output):
+        index, = ctx.saved_tensors
         grad_input = grad_source = None
-        if self.needs_input_grad[0]:
+        if ctx.needs_input_grad[0]:
             grad_input = grad_output.clone()
-            grad_input.scatter_(self.dim, index, 0)
-        if self.needs_input_grad[2]:
-            grad_source = grad_output.gather(self.dim, index)
-        return grad_input, None, grad_source
+            grad_input.scatter_(ctx.dim, index, 0)
+        if ctx.needs_input_grad[3]:
+            grad_source = grad_output.gather(ctx.dim, index)
+        return grad_input, None, None, grad_source, None
 
 
 class Repeat(Function):
 
-    def __init__(self, repeats):
-        super(Repeat, self).__init__()
-        self.repeats = repeats
-
-    def forward(self, input):
-        return input.repeat(self.repeats)
+    @staticmethod
+    def forward(ctx, input, repeats):
+        ctx.repeats = repeats
+        return input.repeat(repeats)
 
-    def backward(self, grad_output):
+    @staticmethod
+    @once_differentiable
+    def backward(ctx, grad_output):
         grad_input = grad_output
-        for dim, repeat in enumerate(self.repeats):
+        for dim, repeat in enumerate(ctx.repeats):
             if repeat == 1:
                 continue
             grad_input = sum(grad_input.chunk(repeat, dim))
-        return grad_input
+        return grad_input, None
 
 
 class Cumsum(Function):
```

Comment:
Because I missed it haha. Anyway, I won't be adding any more new style functions in this PR, I'll leave that for later so that others can contribute too.